Source/Core/Core/PowerPC/Jit64Common/EmuCodeBlock.cpp

```diff
@@ -91,6 +91,16 @@ void EmuCodeBlock::SwitchToNearCode()
   SetCodePtr(m_near_code, m_near_code_end, m_near_code_write_failed);
 }
 
+FixupBranch EmuCodeBlock::BATAddressLookup(X64Reg addr, X64Reg tmp)
+{
+  MOV(64, R(tmp), ImmPtr(&PowerPC::dbat_table[0]));
+  SHR(32, R(addr), Imm8(PowerPC::BAT_INDEX_SHIFT));
+  MOV(32, R(addr), MComplex(tmp, addr, SCALE_4, 0));
```

Comment:
This here compiles to

```
00000122AC983898  mov         rdi,offset PowerPC::dbat_table (07FF6F001F9F0h)  
00000122AC9838A2  shr         edx,11h  
00000122AC9838A5  mov         edx,dword ptr [rdi+rdx*4]  
```

Which seems to be not quite right; you need to clear the high bits of `rdx` somewhere so this is safe, I think? Unless that happens implicitly somewhere? Same thing happens again at the similar construct at `MOV(64, R(tmp), ImmPtr(GetBlockCache()->GetBlockBitSet()))`.

Reply:
Okay, apparently on x64 if you write into a 32-bit register the upper 32 bits of its corresponding 64 bit register get automatically cleared, so this is fine then.